src/sage/schemes/curves/zariski_vankampen.py

```diff
@@ -554,205 +776,165 @@ def populate_roots_interval_cache(inputs):
+            for r in result:
+                roots_interval_cache[r[0][0]] = r[1]
+            problem_par = False
+        except TypeError:
+            pass
 
 
 @parallel
-def braid_in_segment(g, x0, x1):
+def braid_in_segment(glist, x0, x1, precision=dict()):
     """
     Return the braid formed by the `y` roots of ``f`` when `x` moves
     from ``x0`` to ``x1``.
 
     INPUT:
 
-    - ``g`` -- a polynomial factorization in two variables
-    - ``x0`` -- a complex number
-    - ``x1`` -- a complex number
+    - ``glist`` -- a tuple of polynomials in two variables
+    - ``x0`` -- a Gauss rational
+    - ``x1`` -- a Gauss rational
+    - ``precision`` -- a dictionary (default `dict()`) which assigns a number precision bits
+      to each element of ``glist``
 
     OUTPUT:
 
     A braid.
 
     EXAMPLES::
 
-        sage: from sage.schemes.curves.zariski_vankampen import braid_in_segment # optional - sirocco
-        sage: R.<x,y> = QQ[]
+        sage: from sage.schemes.curves.zariski_vankampen import braid_in_segment, fieldI
+        sage: R.<x, y> = QQ[]
+        sage: K = fieldI(QQ)
         sage: f = x^2 + y^3
-        sage: x0 = CC(1,0)
-        sage: x1 = CC(1, 0.5)
-        sage: braid_in_segment(f.factor(), x0, x1) # optional - sirocco
+        sage: f = f.change_ring(K)
+        sage: x0 = 1
+        sage: x1 = 1 + I / 2
+        sage: braid_in_segment(tuple(_[0] for _ in f.factor()), x0, x1)  # optional - sirocco
         s1
 
     TESTS:
 
     Check that :trac:`26503` is fixed::
 
+        sage: # needs sage.rings.real_mpfr sage.symbolic
         sage: wp = QQ['t']([1, 1, 1]).roots(QQbar)[0][0]
         sage: Kw.<wp> = NumberField(wp.minpoly(), embedding=wp)
         sage: R.<x, y> = Kw[]
         sage: z = -wp - 1
-        sage: f = y*(y + z)*x*(x - 1)*(x - y)*(x + z*y - 1)*(x + z*y + wp)
-        sage: from sage.schemes.curves import zariski_vankampen as zvk
-        sage: g = f.subs({x: x + 2*y})
+        sage: f = y * (y + z) * x * (x - 1) * (x - y) * (x + z * y - 1) * (x + z * y + wp)
+        sage: from sage.schemes.curves import zariski_vankampen as zvk  # optional - sirocco
+        sage: Kw1 = zvk.fieldI(Kw)                                      # optional - sirocco
+        sage: g = f.subs({x: x + 2 * y})
+        sage: g = g.change_ring(Kw1)                                    # optional - sirocco
         sage: p1 = QQbar(sqrt(-1/3))
-        sage: p2 = QQbar(1/2+sqrt(-1/3)/2)
-        sage: B = zvk.braid_in_segment(g.factor(),CC(p1),CC(p2)) # optional - sirocco
-        sage: B  # optional - sirocco
+        sage: p1a = CC(p1)
+        sage: p1b = QQ(p1a.real()) + I*QQ(p1a.imag())
+        sage: p2 = QQbar(1/2 + sqrt(-1/3)/2)
+        sage: p2a = CC(p2)
+        sage: p2b = QQ(p2a.real()) + I*QQ(p2a.imag())
+        sage: glist = tuple([_[0] for _ in g.factor()])
+        sage: B = zvk.braid_in_segment(glist, p1b, p2b); B              # optional - sirocco
         s5*s3^-1
```

Comment:
also block-scoped. Strangely `g` depends on `sirocco`, but subsequent lines are not tagged.

Reply:
Actually, only this line needs the tag, I removed the previous ones and changed the import line.

Reply:
OK.